### PR TITLE
chore: enable multiselect in asset-picker network modal

### PR DESCRIPTION
**Pull request by micaelae**
## **Description**
Figma: https://www.figma.com/design/IuOIRmU3wI0IdJIfol0ESu/Cross-Chain-Swaps?node-id=7-24563&node-type=canvas&m=dev

Changes
- enables multi network selection in the Asset Picker, and displaying tokens from multiple chains
- adds support for searching tokens by address
- updates the PickerNetwork component to show multiple network icons
- refactors TokenListItem for less truncation

[Open in GitHub Codespaces](https://codespaces.new/MetaMask/metamask-extension/pull/28975?quickstart=1)

## **Related issues**

Fixes:

## **Manual testing steps**

1. To test locally set `BRIDGE_USE_DEV_APIS=1` in .metamaskrc, then try selecting networks/tokens in the Bridge experience
2. Also try to do some e2e tests in the Send and Swap/Send flows

### **Before -> After**

Screenshot 2024-12-05 at 9 36 14 PMScreenshot 2024-12-05 at 9 34 06 PM

Screenshot 2024-12-05 at 9 36 23 PMScreenshot 2024-12-05 at 9 33 54 PM

Screenshot 2024-12-05 at 9 36 37 PMScreenshot 2024-12-05 at 9 37 45 PM

Screenshot 2024-12-05 at 9 36 44 PMScreenshot 2024-12-05 at 9 37 55 PM

Screenshot 2024-12-10 at 11 25 09 AMScreenshot 2024-12-10 at 11 25 14 AM

## **Pre-merge author checklist**

- [X] I've followed [MetaMask Contributor Docs](https://github.com/MetaMask/contributor-docs) and [MetaMask Extension Coding Standards](https://github.com/MetaMask/metamask-extension/blob/main/.github/guidelines/CODING_GUIDELINES.md).
- [X] I've completed the PR template to the best of my ability
- [X] I’ve included tests if applicable
- [X] I’ve documented my code using [JSDoc](https://jsdoc.app/) format if applicable
- [X] I’ve applied the right labels on the PR (see [labeling guidelines](https://github.com/MetaMask/metamask-extension/blob/main/.github/guidelines/LABELING_GUIDELINES.md)). Not required for external contributors.

## **Pre-merge reviewer checklist**

- [ ] I've manually tested the PR (e.g. pull and build branch, run the app, test code being changed).
- [ ] I confirm that this PR addresses all acceptance criteria described in the ticket it closes and includes the necessary testing evidence such as recordings and or screenshots.

### CODE REVIEW

**Comment by NidhiKJha** [comment]
Nice start but I am not able to select Tokens to bridge to 🤔 
https://github.com/user-attachments/assets/a76c493a-9dce-4c96-9ab4-8fc418a51745

**Review by NidhiKJha — COMMENTED** [review]
I am not able to select Tokens to bridge to 🤔 
https://github.com/user-attachments/assets/a76c493a-9dce-4c96-9ab4-8fc418a51745

File: ui/components/component-library/picker-network/picker-network.tsx [hunk 1]
```diff
@@ -44,20 +46,23 @@ export const PickerNetwork: PickerNetworkComponent = React.forwardRef(
...
+        {avatarComponent ?? (
+          <AvatarNetwork
+            className="mm-picker-network__avatar-network"
+            src={src}
+            name={label}
+            size={AvatarNetworkSize.Xs}
+            {...avatarNetworkProps}
+          />
+        )}
+
```
Comment (NidhiKJha) [inline ui/components/component-library/picker-network/picker-network.tsx:64]: Picker Network is for changing a network so I am not sure why we are making it conditional.

**Review by NidhiKJha — CHANGES_REQUESTED** [review]
Asset Picker seems to be broken in storybook 
Screenshot 2024-12-09 at 11 35 25 AM

File: ui/components/component-library/picker-network/picker-network.tsx (hunk 1 above)
Comment (micaelae) [reply ui/components/component-library/picker-network/picker-network.tsx:63]: Refactored this so an override for the avatar is no longer required. The PickerNetwork for bridge has multiple stacked network icons so we need to be able to render an `AvatarGroup`. Will add a story for this as well

**Review by NidhiKJha — COMMENTED** [review]
Looks really good. Changes I have reviewed:

- [x] verified that the Send flow works as expected
- [x] verified that the asset list shows expected data
- [x] verified that the bridge network picker (excl asset picker) has the expected networks
- [x] AvatarGroup + PickerNetwork styling changes are visible in bridge network picker

Before, I approve. Just one last question: The select all box looks different in Bridge than in other flows:
### Bridge experience select all
Screenshot 2024-12-10 at 5 10 20 PM


### Extension view select all
Screenshot 2024-12-10 at 5 09 26 PM

**Comment by micaelae** [comment]
> Looks really good. Changes I have reviewed:
> 
> * [x]  verified that the Send flow works as expected
> * [x]  verified that the asset list shows expected data
> * [x]  verified that the bridge network picker (excl asset picker) has the expected networks
> * [x]  AvatarGroup + PickerNetwork styling changes are visible in bridge network picker
> 
> Before, I approve. Just one last question: The select all box looks different in Bridge than in other flows:
> 
> ### Bridge experience select all
> Screenshot 2024-12-10 at 5 10 20 PM
> 
> ### Extension view select all
> Screenshot 2024-12-10 at 5 09 26 PM

This is intentional! In a [separate PR](https://github.com/MetaMask/metamask-extension/pull/28976) I'll be updating the shape of the avatar and this size works better for the bridge picker

File: ui/components/component-library/picker-network/picker-network.tsx [hunk 2]
```diff
@@ -27,6 +28,7 @@ export const PickerNetwork: PickerNetworkComponent = React.forwardRef(
   <C extends React.ElementType = 'button'>(
     {
       className = '',
+      avatarGroupProps,
```
Comment (georgewrmarshall) [inline ui/components/component-library/picker-network/picker-network.tsx:31]: Could we add a unit test for this prop to ensure the testing coverage is 100% 🙏

**Review by georgewrmarshall — COMMENTED** [review]
Component Library changes LGTM could we please add a unit test for `avatarGroupProps`

File: ui/components/component-library/picker-network/picker-network.tsx (hunk 2 above)
Comment (micaelae) [reply ui/components/component-library/picker-network/picker-network.tsx:31]: Done!

**Review by matthewwalsh0 — DISMISSED** [review]

**Review by NidhiKJha — DISMISSED** [review]

**Review by georgewrmarshall — DISMISSED** [review]
LGTM!  🚀  Thanks for being so thorough with component library changes 🙏

**Review by georgewrmarshall — DISMISSED** [review]

**Review by georgewrmarshall — APPROVED** [review]

**Review by vinnyhoward — APPROVED** [review]

**Review by sleepytanya — APPROVED** [review]

**Pull request merged**